src/react-components/avatar-selector.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, FormattedMessage } from 'react-intl';
+
+class AvatarSelector extends Component {
+  static propTypes = {
+    avatars: PropTypes.array,
+    avatar: PropTypes.string,
+    onChange: PropTypes.func,
+  }
+
+  getAvatarIndex(direction=0) {
+    const currAvatarIndex = this.props.avatars.findIndex(avatar => avatar.id === this.props.avatar);
+    const numAvatars = this.props.avatars.length;
+    return ((currAvatarIndex + direction) % numAvatars + numAvatars) % numAvatars;
+  }
+
+  nextAvatar = () => {
+    const newAvatarIndex = this.getAvatarIndex(1);
+    this.props.onChange(this.props.avatars[newAvatarIndex].id);
+  }
+
+  prevAvatar = () => {
+    const newAvatarIndex = this.getAvatarIndex(-1);
+    this.props.onChange(this.props.avatars[newAvatarIndex].id);
+  }
+
+  componentDidUpdate() {
+    // HACK - a-animation ought to restart the animation when the `to` attribute changes, but it doesn't
+    // so we need to force it here.
+    const currRot = this.animation.parentNode.getAttribute('rotation');
+    this.animation.setAttribute('from', `${currRot.x} ${currRot.y} ${currRot.z}`);
+    this.animation.stop();
+    this.animation.handleMixinUpdate();
+    this.animation.start();
+  }
+
+  render () {
+    const avatarAssets = this.props.avatars.map(avatar => (
+      <a-progressive-asset
+        id={avatar.id}
+        key={avatar.id}
+        response-type="arraybuffer"
```

Comment:
robert is refactoring things so the semantics here are we have the "normal/default/pbr" asset, and the "unlit" asset, instead of low vs high

Reply:
@robertlong Should I avoid using low/high here?

Reply:
I'm going to leave this as it is until we decide on a final asset bundle format.